### DISCUSSION TITLE
Effects of Updating an Airflow DAG File While It's Running

### DISCUSSION THREAD

**Discussion by maycuatroi1** [apache/airflow#38858] [Q&A]
I would like to ask, with Airflow, when a DAG is running (for about 2-3 hours), and during that time, if I update the DAG file in the dag folder, what will happen?
1. Will the DAG continue to run with the tasks (operators) defined in the old dag?
2. Will it immediately run with the new DAG (I am concerned about synchronization issues here)?

**Comment by TomFaulkner** [comment]
I may be late to this, but it's been my experience that if you replace imported modules it will use the new imported modules on the next task run. I would guess the same is true of the module the DAG is in.
